webapp/sources/rudder/rudder-web/src/main/elm/sources/GroupRelatedRules/ApiCalls.elm

```diff
@@ -0,0 +1,27 @@
+module GroupRelatedRules.ApiCalls exposing (..)
+
+import Http exposing (..)
+import Url.Builder exposing (QueryParameter)
+
+import GroupRelatedRules.DataTypes exposing (..)
+import GroupRelatedRules.JsonDecoder exposing (..)
+
+getUrl: Model -> List String -> List QueryParameter -> String
+getUrl m url p=
+  Url.Builder.relative (m.contextPath :: "secure" :: "api"  :: url) p
+
+getRulesTree : Model -> RelatedRules -> Cmd Msg
+getRulesTree model relatedRules =
+  let
+    req =
+      request
+        { method  = "GET"
+        , headers = []
+        , url     = getUrl model [ "rules", "tree" ] []
```

Comment:
I don't think we want to do that request: it will be huge when there is a lot of rules/directives, when we can have groups with just a few of them. 
And sending big json on the wire is costly. 
We should use a server-side filtered version of that request that only present rules and directives relevant for that group.

Reply:
I thought about adding a list of `ruleIds` to filter the tree on the API side, but I realized we also need to have the info : "Is the current group excluded or included by the each rule" and that blocked me. We [already have that info](https://github.com/Normation/rudder/pull/5297/commits/b602c955a7d570d4d5773e610fd30d7e947c55be#diff-b244376771e5d6dd4b7fa139fc009fa57b16707de2606cfdb1cabd35c817fc3cR222-R244)

So YES !  I guess we could just add the filter on that endpoint now... :smile: